Source/SwiftLintFramework/Rules/OperatorUsageWhitespaceRule.swift

```diff
@@ -78,10 +100,61 @@ public struct OperatorUsageWhitespaceRule: OptInRule, ConfigurationProviderRule
         let kinds = SyntaxKind.commentAndStringKinds()
 
         return file.matchPattern(pattern, excludingSyntaxKinds: kinds,
-                                 excludingPattern: excludingPattern).map {
-            return StyleViolation(ruleDescription: type(of: self).description,
-                                  severity: configuration.severity,
-                                  location: Location(file: file, characterOffset: $0.location))
+                                 excludingPattern: excludingPattern).flatMap {
+
+            let spacesPattern = oneSpace + "*"
+            let rangeRegex = NSRegularExpression
+                .forcePattern(spacesPattern + rangePattern + spacesPattern)
+
+            // if it's a range operator, the correction shouldn't have spaces
+            if let range = rangeRegex.firstMatch(in: file.contents,
+                                                 options: [], range: $0)?.range {
+                let correction = file.contents.bridge().substring(with: range)
+                    .trimmingCharacters(in: .whitespaces)
+                return (range, correction)
+            }
+
+            let pattern = spacesPattern + operators + spacesPattern
+            let operatorsRegex = NSRegularExpression.forcePattern(pattern)
+
+            guard let range = operatorsRegex.firstMatch(in: file.contents,
+                                                        options: [], range: $0)?.range else {
+                return nil
+            }
+
+            let operatorContent = file.contents.bridge().substring(with: range)
+                .trimmingCharacters(in: .whitespaces)
+            let correction = " " + operatorContent + " "
+
+            return (range, correction)
+        }
+    }
+
+    public func correctFile(_ file: File) -> [Correction] {
+        let violatingRanges = violationRanges(file: file).filter { range, _ in
+            return !file.ruleEnabledViolatingRanges([range], forRule: self).isEmpty
+        }
+
+        return writeToFile(file, violatingRanges: violatingRanges)
+    }
+
+    private func writeToFile(_ file: File, violatingRanges: [(NSRange, String)]) -> [Correction] {
+        var correctedContents = file.contents
+        var adjustedLocations = [Int]()
+
+        for (violatingRange, correction) in violatingRanges.reversed() {
+            if let indexRange = correctedContents.nsrangeToIndexRange(violatingRange) {
+                correctedContents = correctedContents
+                    .replacingCharacters(in: indexRange, with: correction)
+                adjustedLocations.insert(violatingRange.location, at: 0)
```

Comment:
Just an FYI, this logic is incorrect. I know it's what I used in yesterday's PR making void return correctable, but this won't successfully adjust the correction locations as characters are removed/added in the file.

It's not a _huge_ deal because we just report these locations in the `swiftlint autocorrect` output log and they're not rendered in Xcode or anything, which is why I was too lazy to do this correctly in my last PR.